### ISSUE TITLE
Address technical debt for data tables

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/fec-cms#3802]
User story: As a developer, it’s very difficult to make changes to data tables because the code is overly complex (?)

:lock: https://docs.google.com/document/d/1E1WlmdzrmPhZMu0X3plaWtN3GRF2bdbbOyv0qruD6Ms/edit# :lock:

**Comment by johnnyporkchops** [comment]
Before committing to a rewrite or refactor of the datatable JS, we should try to anticipate if we would end up with different, but equally complex code, once we include all the necessary controls etc. Arguably, code written by us, however complex of verbose, would be more understandable to us, which is a positive. But it would be great  to also make sure it is improved in being universally readable, editable and extendable.

**Comment by lbeaufort** [comment]
@johnnyporkchops absolutely, I agree. One of the principles of working effectively with legacy code and refactoring is not assuming the code is complex without good reason. My questions above address my overall approach to refactoring. Please feel free to add to the questions if they're missing anything.
